### PR TITLE
IDEMPIERE-5460 - Process Audit doesn't set printing related attribute…

### PR DESCRIPTION
**Pull request by PeterTakacs300**
JIRA ticket: https://idempiere.atlassian.net/browse/IDEMPIERE-5460

# Pull Request Checklist

- [x] My code follows the [code guidelines](https://wiki.idempiere.org/en/Contributing_to_Trunk) of this project
- [x] My code follows the best practices of this project
- [x] I have performed a self-review of my own code
- [x] My code is easy to understand and review. 
- [x] In hard-to-understand areas, I have commented my code.
- [x] My changes generate no new warnings
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have added unit tests that prove my fix is effective or that my feature works
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings
- [ ] I have made corresponding changes to the documentation as follows:
- - [ ] New feature (non-breaking change which adds functionality): I have created the New Feature page in the project wiki explaining the functionality and how to use it. If relevant, I have committed sample data to the core seed to have usable examples in GardenWorld.
- [ ] The changed/added documentation is in the project wiki (not privately-hosted pdf files or links pointing to a company website) and is complete and self-explanatory.

### CODE REVIEW

File: org.adempiere.base/src/org/compiere/print/ReportEngine.java [hunk 1]
```diff
@@ -1838,10 +1839,12 @@ static public ReportEngine get (Properties ctx, ProcessInfo pi, int windowNo)
 		int AD_PrintFormat_ID = 0;
 		boolean IsForm = false;
 		int Client_ID = -1;
+		
+		MPInstance instance = null;
 
 		//	Get AD_Table_ID and TableName
 		StringBuilder sql = new StringBuilder("SELECT rv.AD_ReportView_ID,rv.WhereClause,")
-			.append(" t.AD_Table_ID,t.TableName, pf.AD_PrintFormat_ID, pf.IsForm, pf.AD_Client_ID ")
+			.append(" t.AD_Table_ID,t.TableName, pf.AD_PrintFormat_ID, pf.IsForm, pf.AD_Client_ID, pi.AD_PInstance_ID ")
```
Comment (hengsin) [inline org.adempiere.base/src/org/compiere/print/ReportEngine.java:1847]: This change is not needed, you already have the value from pi.getAD_Pinstance_ID().

File: org.adempiere.base/src/org/compiere/print/ReportEngine.java [hunk 2]
```diff
@@ -1881,7 +1887,7 @@ static public ReportEngine get (Properties ctx, ProcessInfo pi, int windowNo)
 		if (AD_ReportView_ID == 0)
 		{
 			//	Check Print format in Report Directly
-			sql = new StringBuilder("SELECT t.AD_Table_ID,t.TableName, pf.AD_PrintFormat_ID, pf.IsForm ")
+			sql = new StringBuilder("SELECT t.AD_Table_ID,t.TableName, pf.AD_PrintFormat_ID, pf.IsForm, pi.AD_PInstance_ID ")
```
Comment (hengsin) [inline org.adempiere.base/src/org/compiere/print/ReportEngine.java:1890]: Same as above, that's the same value you get from pi.getAD_PInstance_ID()

File: org.adempiere.base/src/org/compiere/print/ReportEngine.java [hunk 3]
```diff
@@ -1915,6 +1922,11 @@ static public ReportEngine get (Properties ctx, ProcessInfo pi, int windowNo)
 					+ ",AD_Client_ID=" + AD_Client_ID);
 				return null;
 			}
+			else
```
Comment (hengsin) [inline org.adempiere.base/src/org/compiere/print/ReportEngine.java:1925]: should make sure instance is not null here, just to be safe.

File: org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/apps/AbstractProcessDialog.java [hunk 4]
```diff
@@ -630,9 +630,12 @@ protected void saveReportOption (){
 		if (!isReport() && !isJasperReport()){
 			return;
 		}
-		if(freportType.getSelectedItem() != null) {
+		if(freportType.getSelectedItem() != null && !Util.isEmpty(freportType.getSelectedItem().getValue())) {
 			getProcessInfo().setReportType(freportType.getSelectedItem().getValue().toString());
 		}
+		else {
+			getProcessInfo().setReportType(MSysConfig.getValue(MSysConfig.ZK_REPORT_TABLE_OUTPUT_TYPE, "HTML"));
```
Comment (hengsin) [inline org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/apps/AbstractProcessDialog.java:637]: hmm ... can it be other report output type here, i.e ZK_REPORT_FORM_OUTPUT_TYPE or ZK_REPORT_FORM_OUTPUT_TYPE ?

Comment (PeterTakacs300) [reply org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/apps/AbstractProcessDialog.java]: According to @norbertbede the these 2 SysConfigs define defaults per form, per table, but they should have the same attributes (HTML, PDF,...). The process audit intention is to catch, which type was taken as default or selected by user. 

In the latest commit I used the logic from ZkReportView.init() for setting the default Report Type

**Review by hengsin — APPROVED** [review]
tested

**Pull request merged**